staging/src/k8s.io/apiserver/pkg/util/openapi/enablement.go

```diff
@@ -45,14 +46,57 @@ func restoreDefinitions(defs map[string]common.OpenAPIDefinition) {
 	if !utilfeature.DefaultFeatureGate.Enabled(genericfeatures.OpenAPIEnums) {
 		for gvk, def := range defs {
 			orig := &def.Schema
-			if ret := pruneEnums(orig); ret != orig {
+
+			ret := pruneEnums(orig)
+			ret = pruneUnrecognizedExtensions(ret)
+
+			if ret != orig {
 				def.Schema = *ret
 				defs[gvk] = def
 			}
 		}
 	}
 }
 
+var whitelistedExtensions sets.String = func() sets.String {
+	res := sets.NewString(
+		"x-kubernetes-list-type",
+		"x-kubernetes-map-keys",
+		"x-kubernetes-map-type",
+		"x-kubernetes-patch-strategy",
+		"x-kubernetes-patch-merge-key",
+	)
+
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.DeclarativeValidationsInOpenAPI) {
+		res.Insert("x-kubernetes-validations")
```

Comment:
This is the CEL extension right? What about all the minimums, maximums etc?

Reply:
Those are already exposed because they are part of the standard OpenAPI spec; so clients should be able to handle them.

Reply:
Yeah, I guess we should remove them if the feature is disabled.

Reply:
This would impact clients who use apiserver as a package. Their OpenAPI would get filtered. Is that something we want to do?

Reply:
You mean apiextension servers? It won't be run against CRDs too will it?

Reply:
No I mean apiserver. Like `sample-apiserver`. This isn't run against CRDs